php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+  public ExpressionTree OBJECT_DIMENSIONAL_LIST() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.OBJECT_DIM_LIST).is(
+      f.objectDimensionalList(
+        b.firstOf(
+          IDENTIFIER(),
+          f.variableName(b.token(PHPLexicalGrammar.KEYWORDS)),
+          COMPUTED_VARIABLE_NAME()),
+        b.zeroOrMore(
+          b.firstOf(
+            ALTERNATIVE_DIMENSIONAL_OFFSET(),
+            DIMENSIONAL_OFFSET()))));
+  }
+
+  public MemberAccessTree CLASS_MEMBER_ACCESS() {
+    return b.<MemberAccessTree>nonterminal(PHPLexicalGrammar.CLASS_MEMBER_ACCESS).is(
+      f.classMemberAccess(
+        b.token(DOUBLECOLON),
+        b.firstOf(
+          VARIABLE_WITHOUT_OBJECTS(),
+          IDENTIFIER(),
+          b.token(CLASS),
+          COMPUTED_VARIABLE_NAME())));
+  }
+
+  public LexicalVariablesTree LEXICAL_VARIABLES() {
+    return b.<LexicalVariablesTree>nonterminal(Kind.LEXICAL_VARIABLES).is(
+      f.lexicalVariables(
+        b.token(USE),
+        b.token(LPARENTHESIS),
+        LEXICAL_VARIABLE(),
+        b.zeroOrMore(f.newTuple11(b.token(COMMA), LEXICAL_VARIABLE())),
+        b.token(RPARENTHESIS)));
+  }
+
+  public VariableTree LEXICAL_VARIABLE() {
+    return b.<VariableTree>nonterminal(PHPLexicalGrammar.LEXICAL_VARIABLE).is(
+      f.lexicalVariable(
+        b.optional(b.token(AMPERSAND)),
+        VARIABLE_IDENTIFIER()));
+  }
+
+  public ExpressionTree STATIC_SCALAR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.STATIC_SCALAR).is(
+      b.firstOf(
+        ARRAY_INITIALIZER(),
+        CONDITIONAL_EXPR()));
+  }
+
+  public FunctionCallTree INTERNAL_FUNCTION() {
+    return b.<FunctionCallTree>nonterminal(PHPLexicalGrammar.INTERNAL_FUNCTION).is(
+      b.firstOf(
+        f.internalFunction1(b.token(PHPLexicalGrammar.ISSET), b.token(LPARENTHESIS), EXPRESSION(), b.zeroOrMore(f.newTuple15(b.token(COMMA), EXPRESSION())), b.token(RPARENTHESIS)),
+
+        f.internalFunction2(
+          b.firstOf(
+            b.token(PHPLexicalGrammar.EMPTY),
+            b.token(PHPLexicalGrammar.EVAL)),
+          b.token(LPARENTHESIS), EXPRESSION(), b.token(RPARENTHESIS)),
+
+        f.internalFunction3(
+          b.firstOf(
+            b.token(PHPLexicalGrammar.INCLUDE_ONCE),
+            b.token(PHPLexicalGrammar.INCLUDE),
+            b.token(PHPLexicalGrammar.REQUIRE_ONCE),
+            b.token(PHPLexicalGrammar.REQUIRE),
+            b.token(PHPLexicalGrammar.CLONE),
+            b.token(PHPLexicalGrammar.PRINT)),
+          EXPRESSION())));
+  }
+
+  public ArrayInitializerTree ARRAY_INITIALIZER() {
+    return b.<ArrayInitializerTree>nonterminal(PHPLexicalGrammar.ARRAY_INIALIZER).is(
+      b.firstOf(
+        f.newArrayInitFunction(b.token(ARRAY), b.token(LPARENTHESIS), b.optional(ARRAY_PAIR_LIST()), b.token(RPARENTHESIS)),
+        f.newArrayInitBracket(b.token(LBRACKET), b.optional(ARRAY_PAIR_LIST()), b.token(RBRACKET))));
+  }
+
+  public SeparatedListImpl<ArrayPairTree> ARRAY_PAIR_LIST() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)